pkg/derivation/derivation.go

```diff
@@ -20,6 +20,50 @@ type Derivation struct {
 	EnvVars          []Env             `json:"env" parser:"'[' ((@@ ','?)* (@@)? )']'')'"`
 }
 
+func (d *Derivation) Validate() error {
+	if len(d.Outputs) == 0 {
+		return fmt.Errorf("at least one output must be defined")
+	}
+
+	for _, o := range d.Outputs {
+		err := o.Validate()
+		if err != nil {
+			return fmt.Errorf("error validating output: %w", err)
+		}
```

Comment:
There's a deterministic way to come up with the hashes of the output path, by looking at the drv.

Check page 94ff of https://edolstra.github.io/pubs/phd-thesis.pdf.

As we don't have the code for doing the hashing yet, we should log a FUTUREWORK here.

Reply:
added FUTUREWORK in 332b5a18f086b9075ea0b5c0da4355c30e1a66c1.